DSharpPlus/Cache/CacheConfiguration.cs

```diff
@@ -0,0 +1,18 @@
+﻿namespace DSharpPlus.Cache;
+
+using System;
+using Microsoft.Extensions.Caching.Memory;
+
+/// <summary>
+/// Configuration for the cache how long to keep items in the cache
+/// </summary>
+/// <remarks>The default implementation of <see cref="IDiscordCache"/> only supports timebased expiration </remarks>
+public record CacheConfiguration
+{
+    public MemoryCacheEntryOptions Guild { get; set; }
+    public MemoryCacheEntryOptions Channel { get; set; }
+    public MemoryCacheEntryOptions Thread { get; set; }
+    public MemoryCacheEntryOptions User { get; set; }
+    public MemoryCacheEntryOptions Message { get; set; }
+    public MemoryCacheEntryOptions Member { get; set; }
+}
```

Comment:
cc @akiraveliara, do we wanna go the more robust route of holding a dictionary of `<Type, CacheExpirations>`? Assuming that this is for V5, and that we have no plans on supporting external cache providers (icky), this should be fine.

Reply:
i do think we want to support external cache providers, so, yes - it doesn't hurt us to try, anyway